src/main/ts/utils/Git.ts

```diff
@@ -117,3 +117,21 @@ const detectGitUrlFromDir = async (dir: string): Promise<string> => {
 
 export const resolveGitUrl = async (gitUrlArg: Option<string>, workingDirArg: string): Promise<string> =>
   O.isSome(gitUrlArg) ? gitUrlArg.value : await detectGitUrlFromDir(workingDirArg);
+
+const isWorkingDirDirty = async (git: SimpleGit) => {
+  const diff = await git.diffSummary([ 'HEAD' ]);
+  return diff.changed > 0;
+};
+
+const isAheadOfRemote = async (git: SimpleGit, branchName: string) => {
+  await git.fetch();
+  const log = await git.log({ from: `origin/${branchName}`, to: branchName, symmetric: false });
+  return log.total > 0;
+};
+
+export const hasLocalChanges = async (workingDir: string, branchName: string) => {
+  const git = gitP(workingDir);
+  const isAhead = await isAheadOfRemote(git, branchName);
+  const isDirty = await isWorkingDirDirty(git);
```

Comment:
Do we still need to check if the working environment is dirty when the branch you've got checked out isn't the branch you're trying to release?

Reply:
Yes, I don't see a downside. This will catch the rare case of someone accidentally on the wrong branch who has made their changes incorrectly